rio_tiler/io/cogeo.py

```diff
@@ -416,7 +416,7 @@ def part(
             **kwargs,
         )
 
-        if expression:
+        if expression and indexes:
```

Comment:
`indexes` will always be defined when using expression because we create `indexes` from `expression` at L402. adding `and indexes` just makes mypy happy